### ISSUE TITLE
Include eslint ignore in file header

### ISSUE THREAD

**Issue by Poimen** [amzn/style-dictionary#677]
We use the standard format outputter `javascript/es6` to emit various amounts of metadata so we can scrape the data for our storybook documentation. However, the output doesn't match our linting requirements.

If I register a new file header processor, I get:
```js
/**
 * eslint-disable
 * Do not edit directly
 * Generated on Wed, 04 Aug 2021 06:24:16 GMT
 */
```

eslint doesn't recognize this (esp. in vscode) and requires something more like:
```js
/* eslint-disable */
/**
 * Do not edit directly
 * Generated on Wed, 04 Aug 2021 06:24:16 GMT
 */
```

I can work around this by re-writing the formatters for outputting the correct file header, then call the built-in formatter to output the data. This feels like a something that could be done internally rather. There could also be a situation for more strict static tools, like Vite, to ignore these files (although I have spent very little time with Vite because it is super strict on some of our other code and couldn't use it in the end). I'm thinking another abstraction on the file header callback where you could provide a list of "lint" disable commands rather than just an array of text to go into the basic header.

Is there any way of getting the eslint disable into the file header correctly without writing my own formatters for built-in versions?

**Comment by dbanksdesign** [comment]
Apologies for the late response, we are going through old issues now...

Unfortunately there is no way to get the eslint disable into the file header correctly without writing a custom formatter. I'm curious, what eslint rules is the generated file failing at? Could also just add the files Style Dictionary generates to the `ignorePatterns` of an eslint config? https://eslint.org/docs/user-guide/configuring/ignoring-code I might also be missing some context too. Let me know if this helps or you have any other questions!

**Comment by Poimen** [comment]
So we've moved on from this issue. Most of the problems we faced with eslint were formatting related and VSCode linting (we don't use prettier because it just not configurable enough and the defaults for us are not good enough - and bike shedding is awesome :D ) - formatting like "file must end with a new line".

Subsequent to this issue, we have found it more useful to use `Object.freeze` on our module exports as there were issues with models being changed unexpectedly, which introduced bugs when the tokens updated. So with `Object.freeze`, the unintended update is prevented. The upshot was that we couldn't use the default outputters anyways and had to write our own - in which we now add the eslint/stylelint disables - like:
```js
function objectToES6String(file, obj) {
  return `/* eslint-disable */${EOL}${StyleDictionary.formatHelpers.fileHeader({ file })}module.exports = Object.freeze(${JSON.stringify(obj, null, 2)});${EOL}`;
}
```

Generated files look:
```js
/* eslint-disable */
/**
 * Do not edit directly
 * Generated on Tue, 26 Oct 2021 06:50:15 GMT
 */
```
and:
```js
/* stylelint-disable */
```
for css.

Seeing that they are generated, if there are any style changes (e.g. 2 vs 4 spaces, tabs vs spaces), then the generator doesn't have to follow as that is not its purpose.

To us, the dev's need to comply with style not generators, so the generators can just push `disabled` comments on their output.

In short, we can close this issue as there is nothing required from us as we've had to move away to custom outputters anyways 👍
